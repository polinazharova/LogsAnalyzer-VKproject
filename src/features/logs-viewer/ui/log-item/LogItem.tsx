import "./LogItem.styles.scss";
import fileIcon from "../../../../assets/file-icon-black.png";

interface LogItemProps {
  fileName: string;
  date: string;
  message: string;
}

const LogItem: React.FC<LogItemProps> = ({ fileName, date, message }) => {
  const parsedDate: Date = new Date(date);

  if (!isNaN(parsedDate.getTime())) {
    date = new Intl.DateTimeFormat("ru-RU", {
      day: "numeric",
      month: "numeric",
      year: "numeric",
      hour: "numeric",
      minute: "numeric",
      second: "numeric",
    }).format(parsedDate);
  }

  return (
    <div className="main__log-viewer-item">
      <img
        className="main__log-viewer-item-file-icon"
        src={fileIcon}
        alt="file icon from flaticon.com"
      />
      <span className="main__log-viewer-item-filename">{fileName}</span>
      <span className="main__log-viewer-item-log-date">{date}</span>
      <span className="main__log-viewer-item-log-message">{message}</span>
    </div>
  );
};

export default LogItem;
