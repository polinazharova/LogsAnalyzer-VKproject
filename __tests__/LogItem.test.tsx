import React from "react";
import LogItem from "../src/features/logs-viewer/ui/log-item/LogItem";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

const props: { [key: string]: string } = {
  fileName: "test.log",
  date: "13.02.2003",
  message: "test",
};

describe("LogItem component", () => {
  test("renders component correctly", () => {
    render(
      <LogItem
        fileName={props.fileName}
        date={props.date}
        message={props.message}
      />,
    );
    const logItem = document.getElementsByClassName("main__log-viewer-item")[0];
    expect(logItem).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(
      <LogItem
        fileName={props.fileName}
        date={props.date}
        message={props.message}
      />,
    );
    expect(container).not.toBeEmptyDOMElement();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("applies correct data", () => {
    const date = "2025-02-12T19:36:10+03:00";
    render(
      <LogItem fileName={props.fileName} date={date} message={props.message} />,
    );
    expect(screen.getByText(props.fileName)).toBeInTheDocument();
    expect(screen.getByText("12.02.2025, 19:36:10")).toBeInTheDocument();
    expect(screen.getByText(props.message)).toBeInTheDocument();
  });

  test("renders more than 1 component correctly", () => {
    render(
      <LogItem
        fileName={props.fileName}
        date={props.date}
        message={props.message}
      />,
    );
    render(
      <LogItem
        fileName={props.fileName}
        date={props.date}
        message={props.message}
      />,
    );
    render(
      <LogItem
        fileName={props.fileName}
        date={props.date}
        message={props.message}
      />,
    );
    const logItems = document.getElementsByClassName("main__log-viewer-item");
    expect(logItems).toHaveLength(3);
  });
});
