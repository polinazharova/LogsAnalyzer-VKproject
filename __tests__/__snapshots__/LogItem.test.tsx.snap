// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogItem component matches snapshot 1`] = `
<div
  class="main__log-viewer-item"
>
  <img
    alt="file icon from flaticon.com"
    class="main__log-viewer-item-file-icon"
    src=""
  />
  <span
    class="main__log-viewer-item-filename"
  >
    test.log
  </span>
  <span
    class="main__log-viewer-item-log-date"
  >
    NaN
  </span>
  <span
    class="main__log-viewer-item-log-message"
  >
    test
  </span>
</div>
`;
